### ISSUE TITLE
ETARGET for TypeScript 3.3

### ISSUE THREAD

**Issue by buschtoens** [Microsoft/dtslint#166]
Currently all our CI jobs fails, because `dtslint` tries to install `typescript@3.3`, which is not released yet. This is because it has been added to [`TypeScriptVersion.all` in `definitelytyped-header-parser`](https://github.com/Microsoft/definitelytyped-header-parser/blob/49748ca5b2dc84563b31e26528c725d12e6668fd/index.ts#L16-L18).

```ts
export namespace TypeScriptVersion {
	export const all: ReadonlyArray<TypeScriptVersion> =
		["2.0", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9", "3.0", "3.1", "3.2", "3.3"];
	// ...
}
```

This causes `installAll` to choke on L13.

https://github.com/Microsoft/dtslint/blob/c26894e43837919306878ecb0562f491e3a9cf24/src/installer.ts#L9-L16

```
lerna ERR! npm ERR! Linux 4.4.0-101-generic
lerna ERR! npm ERR! argv "/home/travis/.nvm/versions/node/v6.15.1/bin/node" "/home/travis/.nvm/versions/node/v6.15.1/bin/npm" "install" "--ignore-scripts" "--no-shrinkwrap" "--no-package-lock" "--no-bin-links"
lerna ERR! npm ERR! node v6.15.1
lerna ERR! npm ERR! npm  v3.10.10
lerna ERR! npm ERR! code ETARGET
lerna ERR! 
lerna ERR! npm ERR! notarget No compatible version found: typescript@3.3
lerna ERR! npm ERR! notarget Valid install targets:
lerna ERR! npm ERR! notarget 3.3.0-dev.20181211, 3.3.0-dev.20181208, 3.3.0-dev.20181207, ...
```

Full log on [Travis CI](https://travis-ci.org/ember-decorators/ember-decorators/jobs/466651153).

**Comment by buschtoens** [comment]
If you are using yarn, you can circumvent this issue by forcing a resolution for the [latest compatible version of `definitelytyped-header-parser`](https://github.com/Microsoft/definitelytyped-header-parser/commit/4941b70b7658d5238b619dcba694aee22f5b8f56).

You can find an exemplary PR here: https://github.com/ember-decorators/ember-decorators/pull/338

**Comment by buschtoens** [comment]
Hold on one second. L12 is supposed to prevent this from happening in the first place. 🤔

**Comment by buschtoens** [comment]
OMG. 🤦‍♂️ 

We were using an outdated version of `dtslint`, that didn't have the check yet. Sorry for the noise.

<sup>It's a bit annoying though, that his was possible to happen in the first place. 😅 </sup>
